Formula/cp2k.rb

```diff
@@ -1,10 +1,9 @@
 class Cp2k < Formula
   desc "Quantum chemistry and solid state physics software package"
   homepage "https://www.cp2k.org/"
-  url "https://github.com/cp2k/cp2k/releases/download/v8.2.0/cp2k-8.2.tar.bz2"
-  sha256 "2e24768720efed1a5a4a58e83e2aca502cd8b95544c21695eb0de71ed652f20a"
+  url "https://github.com/cp2k/cp2k/releases/download/v2022.1/cp2k-2022.1.tar.bz2"
```

Comment:
We should double-check that upstream have changed their version scheme permanently. It'll be very annoying to change it back if they decide to go back to the old one.

Reply:
It looks like it based on resources/comments available.

Last meeting notes https://www.cp2k.org/_media/dev:cp2k_dev_meeting_05_05_22.pdf
> - switching to year.month or year.number naming schema?
> - yes, (year).(release number in year)
 
Google groups release https://groups.google.com/g/cp2k/c/CjSsK2BX8LQ/m/ne2ypAu-AAAJ

And comment in another project https://github.com/aiidateam/aiida-cp2k/issues/163#issuecomment-1181747449
> The pace of development in CP2K has notably picked up over the last years. Therefore we now adopted a 6 month release cycle. Alongside, we changed our versioning scheme from `<major>.<minor>` to `<year>.<number>` to make it more self-explanatory.